### PR TITLE
Basic usage with affine transform and a single regression test

### PR DESCRIPTION
**Pull request by dzenanz**

### CODE REVIEW

File: include/itkANTSRegistration.h [hunk 1]
```diff
@@ -138,6 +156,8 @@ class ANTSRegistration : public ProcessObject
 
   std::string m_TypeOfTransform{ "Affine" };
 
+  typename RegistrationHelperType::Pointer m_Helper{ RegistrationHelperType::New() };
```
Comment (thewtex) [inline include/itkANTSRegistration.h:159]: `private`

File: include/itkANTSRegistration.hxx [hunk 2]
```diff
@@ -129,20 +173,122 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::AllocateOutpu
...
+  outputImage->DisconnectPipeline();
+  return outputImage;
+}
+
 template <typename TFixedImage, typename TMovingImage, typename TParametersValueType>
 void
 ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::GenerateData()
 {
-  this->AllocateOutputs();
+  // this->AllocateOutputs();
```
Comment (thewtex) [inline include/itkANTSRegistration.hxx:204]: uncomment

File: include/itkANTSRegistration.hxx [hunk 3]
```diff
@@ -129,20 +173,122 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::AllocateOutpu
...
+
+  std::stringstream ss;
+  m_Helper->SetLogStream(ss);
+
+  std::string whichTransform = this->GetTypeOfTransform();
```
Comment (thewtex) [inline include/itkANTSRegistration.hxx:209]: -> behavior of ANTs with possibly multiple transforms

File: include/itkANTSRegistration.hxx [hunk 4]
```diff
@@ -129,20 +173,122 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::AllocateOutpu
...
+
+  int retVal = m_Helper->DoRegistration();
+  if (retVal != EXIT_SUCCESS)
+  {
+    itkExceptionMacro(<< "Registration failed. Helper's accumulated output:\n " << ss.str());
+  }
+
+  typename DecoratedOutputTransformType::Pointer decoratedOutputTransform = DecoratedOutputTransformType::New();
+  decoratedOutputTransform->Set(m_Helper->GetModifiableCompositeTransform());
+  this->SetForwardTransformInput(decoratedOutputTransform);
```
Comment (thewtex) [inline include/itkANTSRegistration.hxx:291]: Also output the inverse transform

**Comment by dzenanz** [comment]
This is now down to MacOS compile warnings:
https://open.cdash.org/viewBuildError.php?type=1&buildid=9322560

The first one is:
```log
[24/26] Building CXX object test/CMakeFiles/ANTsWasmTestDriver.dir/itkANTSRegistrationBasicTests.cxx.o
In file included from test/itkANTSRegistrationBasicTests.cxx:19:
In file included from include/itkANTSRegistration.h:25:
In file included from _deps/ants_inner-src/Examples/itkantsRegistrationHelper.h:80:
In file included from Modules/Registration/RegistrationMethodsv4/include/itkTimeVaryingBSplineVelocityFieldImageRegistrationMethod.h:25:
In file included from Modules/Filtering/DisplacementField/include/itkTimeVaryingBSplineVelocityFieldTransform.h:197:
In file included from Modules/Filtering/DisplacementField/include/itkTimeVaryingBSplineVelocityFieldTransform.hxx:26:
In file included from Modules/Filtering/DisplacementField/include/itkTimeVaryingVelocityFieldIntegrationImageFilter.h:198:

Modules/Filtering/DisplacementField/include/itkTimeVaryingVelocityFieldIntegrationImageFilter.hxx:140:98: warning: instantiation of variable 'itk::NumericTraits<itk::Vector<double, 2>>::Zero' required here, but no definition is available [-Wundefined-var-template]

    this->GetOutput()->FillBuffer(itk::NumericTraits<typename DisplacementFieldType::PixelType>::Zero);
                                                                                                 ^
```

**Comment by dzenanz** [comment]
Visual Studio with clang backend reproduces these warnings, which means I should be able to take care of them without issues.

File: include/itkANTSRegistration.hxx (hunk 3 above)
Comment (dzenanz) [reply include/itkANTSRegistration.hxx:220]: We later agreed to do one transform at a time. If multiple stages are needed, solve it via setting the initial transform.

Comment (thewtex) [reply include/itkANTSRegistration.hxx:220]: To be clear, we discussed following the behavior of [antspyx' python interface](https://antspyx.readthedocs.io/en/latest/registration.html), which has some pre-defined multi-transform optimizations, e.g. `"TRSAA"`, which are registered in call. But additional, multi-stage registrations would be made via multiple calls.

File: include/itkANTSRegistration.hxx [hunk 5]
```diff
@@ -129,20 +173,146 @@ ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::AllocateOutpu
...
-//  -> DataObjectPointer
+// template <typename TFixedImage, typename TMovingImage, typename TParametersValueType>
+// auto
+// ANTSRegistration<TFixedImage, TMovingImage, TParametersValueType>::MakeOutput(DataObjectPointerArraySizeType)
+//   -> DataObjectPointer
 //{
-//  return DataObjectPointer();
-//}
+//   return DataObjectPointer();
+// }
```
Comment (thewtex) [inline include/itkANTSRegistration.hxx:182]: Implement

**Review by thewtex — CHANGES_REQUESTED** [review]
Looks good :+1: 

Please implement `MakeOutput`.

**Review by thewtex — APPROVED** [review]
💯

**Pull request merged**